injected/src/features/broker-protection/utils/expectations.js

```diff
@@ -0,0 +1,123 @@
+import { getElement } from "./utils.js";
+
+/**
+ * Return a true/false result for every expectation
+ *
+ * @param {import("../types").Expectation[]} expectations
+ * @param {Document | HTMLElement} root
+ * @return {import("../types").BooleanResult[]}
+ */
+export function expectMany(expectations, root) {
+    return expectations.map((expectation) => {
+        switch (expectation.type) {
+            case 'element':
+                return elementExpectation(expectation, root);
+            case 'text':
+                return textExpectation(expectation, root);
+            case 'url':
+                return urlExpectation(expectation);
+            default: {
+                return {
+                    result: false,
+                    error: `unknown expectation type: ${expectation.type}`,
+                };
+            }
+        }
+    });
+}
+
+/**
+ * Verify that an element exists. If the `.parent` property exists,
+ * scroll it into view first
+ *
+ * @param {import("../types").Expectation} expectation
+ * @param {Document | HTMLElement} root
+ * @return {import("../types").BooleanResult}
+ */
+export function elementExpectation(expectation, root) {
+    if (expectation.parent) {
+        const parent = getElement(root, expectation.parent);
+        if (!parent) {
+            return {
+                result: false,
+                error: `parent element not found with selector: ${expectation.parent}`,
+            };
+        }
+        parent.scrollIntoView();
```

Comment:
Wouldn't keeping conditionals isolated from actions be cleaner? Again we could then reuse these conditions elsewhere in C-S-S more.

Reply:
Happy to discuss further when we work on consolidating with the other related code in C-S-S.